core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,594 @@
+    #
+    #   daily_param_stats : { param_name1 : { param_value1 : report_count1,
+    #                                         param_value2 : report_count2,
+    #                                         param_value3 : report_count3 },
+    #                         param_name2 : { param_value1 : report_count1,
+    #                                         param_value2 : report_count2,
+    #                                         param_value3 : report_count3 } }
+    #
+    # The second key in the JSON -- daily_total_reports_submitted -- simply has
+    # the total number of reports submitted on this date:
+    #
+    #   daily_total_reports_submitted : total_reports_count.
+    daily_ticket_stats = datastore_services.JsonProperty(
+        required=True, indexed=False)
+    # The schema version for parameter statistics in this entity.
+    daily_ticket_stats_schema_version = datastore_services.IntegerProperty(
+        required=True, indexed=True)
+
+    @classmethod
+    def create(
+            cls, platform, ticket_id, stats_tracking_date, daily_ticket_stats):
+        """Creates a new AppFeedbackReportStatsModel instance and returns its
+        ID.
+
+        Args:
+            ticket_id: str. The ID for the ticket these stats aggregate on.
+            platform: str. The platform the stats are aggregating for.
+            stats_tracking_date: datetime.date. The date in UTC that this entity
+                is tracking stats for.
+            daily_ticket_stats: dict. The daily stats for this entity, keyed
+                by the parameter witch each value mapping a parameter value to
+                the number of reports that satisfy that parameter value.
+
+        Returns:
+            AppFeedbackReportStatsModel. The newly created
+            AppFeedbackReportStatsModel instance.
+        """
+        entity_id = cls._generate_id(platform, ticket_id, stats_tracking_date)
+        stats_entity = cls(
+            id=entity_id, ticket_id=ticket_id, platform=platform,
+            stats_tracking_date=stats_tracking_date,
+            daily_ticket_stats=daily_ticket_stats,
+            daily_ticket_stats_schema_version=(
+                feconf.CURRENT_REPORT_STATS_SCHEMA_VERSION))
+        stats_entity.update_timestamps()
+        stats_entity.put()
+        return entity_id
+
+    @classmethod
+    def _generate_id(cls, platform, ticket_id, stats_tracking_date):
+        """Generates key for the instance of AppFeedbackReportStatsModel
+        class in the required format with the arguments provided.
+
+        Args:
+            platform: str. The platform this entity is aggregating on.
+            ticket_id: str. The ID for the ticket these stats aggregate on.
+            stats_tracking_date: date. The date these stats are tracking on.
+
+        Returns:
+            str. The generated ID for this entity of the form
+            '[platform]:[ticket_id]:[date.seconds]'.
```

Comment:
Here, you say date.seconds (i.e. the "seconds" attribute of the date) but in the implementation you seem to be using date.isoformat() which is a completely different thing...

Reply:
Ah forgot to update the docstring,  I had used isoformat here since date.seconds wasn't possible. Using isoformat() in all these cases instead now since timestamp() (which gives the date / datetime in seconds) is not available in Python 2.

Reply:
After a follow-up, will use milliseconds to match current convention for other entity_ids.